cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/SparkCqlEvaluator.java

```diff
@@ -300,17 +311,36 @@ public void run(PrintStream out) throws Exception {
 
                     CustomMetricSparkPlugin.currentlyEvaluatingContextGauge.setValue(CustomMetricSparkPlugin.currentlyEvaluatingContextGauge.getValue() + 1);
                     JavaPairRDD<Object, Map<String, Object>> resultsByContext = rowsByContextId
-                            .mapToPair(x -> evaluate(contextName, x, perContextAccum));
+                            .mapToPair(x -> evaluate(contextName, x, perContextAccum, errorAccumulator));
                     
                     writeResults(spark, resultsSchema, resultsByContext, outputPath);
 
                     LOG.info(String.format("Wrote results for context %s to %s", contextName, outputPath));
 
+                    evaluationSummary.addContextCount(contextName, perContextAccum.value());
+                    
                     contextAccum.add(1);
                     perContextAccum.reset();
                 }
             }
+            
+            evaluationSummary.setEndTimeMillis(System.currentTimeMillis());
+
+            if (args.metadataOutputPath != null) {
+                Path metadataPath = new Path(args.metadataOutputPath);
+                
+                if (errorAccumulator != null) {
+                    evaluationSummary.setErrorList(errorAccumulator.value());
+                }
+                
+                evaluationSummary.setTotalContexts(contextAccum.value());
+                
+                HadoopPathOutputMetadataWriter writer = new HadoopPathOutputMetadataWriter(metadataPath, hadoopConfiguration.value());
```

Comment:
The LHS of this expression should be OutputMetadataWriter and I'd like to see the RHS be a function call on this class `getOutputMetadataWriter()` so that people can override the behavior if they want.